runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -1368,9 +1369,10 @@ J9::ARM64::TreeEvaluator::VMnewEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    // 5. Allocate object/array on heap
    genHeapAlloc(node, cg, isVariableLength, allocateSize, elementSize, resultReg, lengthReg, tempReg1, tempReg2, tempReg3, conditions, callLabel);
 
-   // 6. Setup HeapAllocSnippet for slowpath
-   TR::Snippet *snippet = new (cg->trHeapMemory()) TR::ARM64HeapAllocSnippet(cg, node, callLabel, node->getSymbolReference(), doneLabel);
-   cg->addSnippet(snippet);
+   // 6. Setup OOL Section for slowpath
+   TR::Register *objReg = cg->allocateCollectedReferenceRegister();
+   TR_ARM64OutOfLineCodeSection *outlinedHelperCall = new (cg->trHeapMemory()) TR_ARM64OutOfLineCodeSection(node, TR::acall, objReg, callLabel, callReturnLabel, cg);
```

Comment:
You use `objReg` as the target register of the OOL section here.
What is the reason of calling `generateMovInstruction(cg, node, objReg, resultReg, true);` later?

Reply:
The allocated object is stored in `objReg`, so `objReg` is a collected reference register. `resultReg` is a temporary register that holds the address of the object, but it must *not* be a collected reference register. In our instruction stream, `resultReg` is already used before branching into the OOL section. If it is a collected reference register, it will be added to the GC map at the branch instruction to the helper in the OOL section. At this point, the value in `resultReg` is not a valid object, so `resultReg` must not be added to the GC map. 
`resultReg` is copied to `objReg` after all checks whether the slow path is taken are done.

Reply:
Thank you for explaining.  I understand.